src/backend/distributed/serialize_distributed_ddls.c

```diff
@@ -0,0 +1,70 @@
+/*-------------------------------------------------------------------------
+ *
+ * serialize_distributed_ddls.c
+ *
+ * This file contains functions for serializing distributed DDLs.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "miscadmin.h"
+
+#include "catalog/dependency.h"
+#include "storage/lock.h"
+
+#include "distributed/adaptive_executor.h"
+#include "distributed/deparse_shard_query.h"
+#include "distributed/metadata_sync.h"
+#include "distributed/resource_lock.h"
+#include "distributed/serialize_distributed_ddls.h"
+
+
+/*
+ * SerializeDistributedDDLsOnObjectClass serializes distributed DDLs that
+ * target given object class by acquiring a Citus specific advisory lock
+ * on the first primary worker node.
+ *
+ * The lock is acquired via a coordinated transaction. For this reason,
+ * it automatically gets released when (maybe implicit) transaction on
+ * current server commits or rolls back.
```

Comment:
Verify if that's really the case for create database.

Do we by mistake commit right after SET LOCAL thing?

Reply:
no